extensions/test/kokkos/spaces.cpp

```diff
@@ -0,0 +1,73 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <Kokkos_Core.hpp>
+
+
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/extensions/kokkos/spaces.hpp>
+
+
+#include "core/test/gtest/environments.hpp"
+#include "core/test/utils.hpp"
+
+
+TEST(Executor, CanCreateDefaultExecutor)
+{
+    auto exec = gko::ext::kokkos::create_default_executor();
+
+#if defined(KOKKOS_ENABLE_CUDA)
+    ASSERT_TRUE(std::dynamic_pointer_cast<gko::CudaExecutor>(exec));
+#elif defined(KOKKOS_ENABLE_HIP)
+    ASSERT_TRUE(std::dynamic_pointer_cast<gko::HipExecutor>(exec));
+#elif defined(KOKKOS_ENABLE_SYCL)
+    ASSERT_TRUE(std::dynamic_pointer_cast<gko::DpcppExecutor>(exec));
+#elif defined(KOKKOS_ENABLE_OPENMP)
+    // necessary because of our executor hierarchy...
+    ASSERT_FALSE(std::dynamic_pointer_cast<gko::ReferenceExecutor>(exec));
+    ASSERT_TRUE(std::dynamic_pointer_cast<gko::OmpExecutor>(exec));
+#elif defined(KOKKOS_ENABLE_SERIAL)
+    ASSERT_TRUE(std::dynamic_pointer_cast<gko::ReferenceExecutor>(exec));
+#else
+    ASSERT_TRUE(false);
+#endif
+}
+
+
+TEST(Executor, CanCreateExecutorWithExecutorSpace)
+{
+#ifdef KOKKOS_ENABLE_SERIAL
+    auto exec = gko::ext::kokkos::create_executor(Kokkos::Serial{});
+
+    ASSERT_TRUE(std::dynamic_pointer_cast<gko::ReferenceExecutor>(exec));
+#endif
```

Comment:
also omp part?
I assume there's always serial or omp as the host part (other than accelerator)

Reply:
I only need to check one execution space, since I just want to test explicitly choosing the executor. I don't think there is any need for a second execution space.
Also, for Kokkos you can't assume that omp is available, they might use threads instead for the host.

Reply:
the test may not have something to do when the kokkos does not compile for serial.

Reply:
Yes, but I think we will be able to control how Kokkos is setup for our tests, so we don't need to check for each possible host executor.